### DISCUSSION TITLE
有没有什么内存开销小一些的wireguard使用方式？

### DISCUSSION THREAD

**Discussion by hahach** [XTLS/Xray-core#3444] [Q&A]
我有一台openvz小鸡，ipv6流量无限，ipv4流量有限制。所以在该鸡做了点不太常见的使用设置。

域名绑了ipv6，使用ipv6入站，出站加了CF的warp，通过ipv6连接。本地搭建了mosdns优先解析域名为ipv6。xray的路由规则把所有ipv4全都路由到warp，ipv6地址的都是正常放行。

使用起来也没什么毛病，就是wireguard协议是不是太费内存了。512m的小鸡在走wireguard出站的时候跑过200mbps时很容易让xray内存占用跑上300到400M，导致 OOM。尝试过wireproxy，也是很费内存。不知道有没有其他什么wireguard的实现能省点内存的？

尝试过把warp出站替换为CF worker搭建的vless节点，小鸡解放了，但在高速访问的时候也容易让worker猝死。
或者佬们有没有什么其他的知道的可以薅来当做ipv6中转的服务？

**Comment by hahach** [comment]
绷不住了，隔了半年，试了下cloudflare官方的warp-cli，发现官方家伙才是省内存省cpu的终极方案。分流交给xray，warp-cli采用proxy mode，开个socks给xray，再也不用担心内存爆炸了

**Reply by Ghost3972** [reply]
看到之后我也去试了一下，结果cloud flare client不支持alpine，草。。
